pkg/controller/workload/job/job_controller.go

```diff
@@ -477,9 +477,12 @@ func podsCount(jobSpec *batchv1.JobSpec) int32 {
 func generatePodsReadyCondition(job *batchv1.Job, wl *kueue.Workload) metav1.Condition {
 	conditionStatus := metav1.ConditionFalse
 	message := "Not all pods are ready or succeeded"
-	if podsReady(job) && wl.Spec.Admission != nil {
+	// Once PodsReady=True it stays as long as the workload remains admitted to
+	// avoid unnecessary flickering the the condition when the pods transition
+	// Ready to Completed.
```

Comment:
Extra clarification:
As pods finish, they transition first into the `uncountedTerminatedPods` staging area, before passing to the succeeded/failed counters

Reply:
Which brings us to an alternative:
count the pods in the staging field. But maybe that's overkill.